src/cpp/session/include/session/projects/SessionProjects.hpp

```diff
@@ -152,6 +152,8 @@ class ProjectContext : boost::noncopyable
    {
       return packageInfo_;
    }
+   
+   bool isPackageProject();
```

Comment:
Does our async infer code work for package source files NOT in the currently active R package project? For folks like Hadley that have a lot of packages which relate it seems like it would be common to be browsing R source code in another package and therefore important that this works outside of the context of projects. There are also people (shudder) who build R packages without RStudio projects (they just use devtools or other ad-hoc methods).